experimental/iterators/include/iterators/Dialect/Iterators/IR/IteratorsOps.td

```diff
@@ -303,4 +303,90 @@ def Iterators_SinkOp : Iterators_Base_Op<"sink"> {
   let arguments = (ins Iterators_StreamOfLLVMStructOfNumerics:$input);
 }
 
+//===----------------------------------------------------------------------===//
+// Ops related to Iterator bodies.
+//===----------------------------------------------------------------------===//
+
+def Iterators_UndefStateOp : Iterators_Base_Op<"undefstate", [NoSideEffect]> {
+  let summary = "Create an undefined iterator state";
+  let results = (outs Iterators_State:$result);
+  let assemblyFormat = "attr-dict `:` type($result)";
+  let description = [{
+    Creates an iterator state of the given type with undefined field values.
```

Comment:
Please take a look at this as background info: https://discourse.llvm.org/t/rfc-poison-semantics-for-mlir/66245